public/pages/Rollups/containers/Rollups/Rollups.tsx

```diff
@@ -0,0 +1,583 @@
+      history.replace({ ...this.props.location, search: queryParamsString });
+      const rollupJobsResponse = await rollupService.getRollups(queryParamsString);
+      if (rollupJobsResponse.ok) {
+        const { rollups, totalRollups } = rollupJobsResponse.response;
+        this.setState({ rollups, totalRollups });
+      } else {
+        toastNotifications.addDanger(rollupJobsResponse.error);
+      }
+    } catch (err) {
+      toastNotifications.addDanger(getErrorMessage(err, "There was a problem loading the rollups"));
+    }
+    this.setState({ loadingRollups: false });
+  };
+
+  getExplains = async (): Promise<void> => {
+    this.setState({ loadingRollups: true });
+    try {
+      const { rollupService } = this.props;
+      const { rollups } = this.state;
+      //Concat the rollup job ids to form params for explain api:
+      let paramString = "";
+      rollups.map((rollup) => {
+        if (rollups.indexOf(rollup) == 0) {
+          paramString = paramString + rollup._id;
+        } else {
+          paramString = paramString + "," + rollup._id;
+        }
+      });
+      const rollupJobsResponse = await rollupService.explainRollup(paramString);
+      if (rollupJobsResponse.ok) {
+        const rollupExplain = rollupJobsResponse.response;
+        this.setState({ rollupExplain });
+        // Assuming the order of rollup jobs is identical to metadata.
+        rollups.map((rollup: RollupItem) => {
+          rollup.rollup.metadata = rollupExplain[rollup._id];
+        });
+        this.setState({ rollups });
+      } else {
+        toastNotifications.addDanger(rollupJobsResponse.error);
+      }
+    } catch (err) {
+      toastNotifications.addDanger(getErrorMessage(err, "There was a problem loading the metadata of rollups"));
+    }
+    this.setState({ loadingRollups: false });
+  };
+
+  onClickCreate = (): void => {
+    this.props.history.push(ROUTES.CREATE_ROLLUP);
+  };
+
+  onClickEdit = (): void => {
+    const {
+      selectedItems: [{ _id }],
+    } = this.state;
+    if (_id) this.props.history.push(`${ROUTES.EDIT_ROLLUP}?id=${_id}`);
+  };
+
+  onClickDelete = async (): Promise<void> => {
+    const { rollupService } = this.props;
+    const { selectedItems } = this.state;
+    var item;
+    for (item of selectedItems) {
+      const rollupId = item._id;
+      try {
+        const response = await rollupService.deleteRollup(rollupId);
+
+        if (response.ok) {
+          this.closeDeleteModal();
+          await this.getRollups();
+          //Show success message
+          toastNotifications.addSuccess(`"${rollupId}" successfully deleted!`);
+        } else {
+          toastNotifications.addDanger(`Could not delete the rollup job "${rollupId}" : ${response.error}`);
+        }
+      } catch (err) {
+        toastNotifications.addDanger(getErrorMessage(err, "Could not delete the rollup job"));
+      }
+    }
+  };
+
+  onDisable = async (): Promise<void> => {
```

Comment:
This comment applies to all onClickDelete, onDisable, onEnable

We allow people to select a large # of rollups, especially when changing the pagination to the maximum. This code is set up to iterate over every item in the selection and do a request to delete, stop, or start, and then also get all rollups and get explains (for disable and enable) and then add a toast message which could end up showing 100 toaster messages all very quickly (or slowly as going through 50 or 100 of these could take a long time).

At minimum:
Move the getRollups/getExplain out of the loop and do it only at the end.
Consider having only one toaster at the end with an aggregated response of successful/fail.

Ideally:
Send a single request to nodejs with all rollupIds and do the operations there and then on browser once that request is done you can do the getROllups/getExplains (ideally also as one request).

Reply:
Moved `getRollups/getExplain `out of the loop for now. Will look into solution to send single request.